src/app/snapshot/snapshots-page/snapshot-list-item/snapshot-item.component.ts

```diff
@@ -0,0 +1,72 @@
+import { EventEmitter } from '@angular/core';
+import { MatMenuTrigger } from '@angular/material';
+import { Dictionary } from '@ngrx/entity/src/models';
+import { TranslateService } from '@ngx-translate/core';
+import {
+  getDateSnapshotCreated,
+  Snapshot,
+  SnapshotStates,
+  Volume
+} from '../../../shared/models';
+import { VirtualMachine } from '../../../vm';
+
+export class SnapshotItemComponent {
+  public item: Snapshot;
+  public volumes: Dictionary<Volume>;
+  public virtualMachines: Dictionary<VirtualMachine>;
+  public isSelected: (snapshot: Snapshot) => boolean;
+  public onClick = new EventEmitter<Snapshot>();
+  public matMenuTrigger: MatMenuTrigger;
+
+  public stateTranslations = {
+    [SnapshotStates.BackedUp]: 'SNAPSHOT_STATE.BACKEDUP',
+    [SnapshotStates.BackingUp]: 'SNAPSHOT_STATE.BACKINGUP',
+    [SnapshotStates.Creating]: 'SNAPSHOT_STATE.CREATING',
+    [SnapshotStates.Allocated]: 'SNAPSHOT_STATE.ALLOCATED',
+    [SnapshotStates.Error]: 'SNAPSHOT_STATE.ERROR',
+  };
+
+  public get statusClass() {
+    const { state } = this.item;
+    const backedUp = state === SnapshotStates.BackedUp;
```

Comment:
very strange approach. please consider modifying the following flags initialization.
map could be used